projects/notebooks/testing/generate_matrix-benchmarking.sh

```diff
@@ -21,309 +21,310 @@ export MATBENCH_WORKLOAD=projects.notebooks.visualizations.$(get_config matbench
-            else
-                _info "An error code #$regression_analyses_retcode happened during the regression analyses, but the 'fail_test_on_regression_fail' flag is not set. Ignoring."
-            fi
-        fi
-
-        rm -f .env.yaml
-    fi
-
-    #
-    # Upload LTS results
-    #
-
-    if test_config matbench.lts.opensearch.upload; then
-        generate_opensearch_config
-
-        step_idx=$((step_idx + 1))
-        if ! matbench upload_lts |& tee > "$ARTIFACT_DIR/${step_idx}_matbench_upload_lts.log"; then
-            _warning "An error happened while uploading the LTS payload :/"
-            if test_config matbench.lts.opensearch.fail_test_on_upload_fail; then
-                retcode=1
-            fi
-        fi
-        rm -f .env.yaml
-    fi
-
-    if test_config matbench.download.save_to_artifacts; then
-        cp -rv "$MATBENCH_RESULTS_DIRNAME" "$ARTIFACT_DIR"
-    fi
-
-    #
-    # Generate the visualization reports
-    #
-
-    for filters_to_apply in $filters; do
-        if [[ "$filters_to_apply" == "null" ]]; then
-            filters_to_apply=""
-        fi
-
-        mkdir -p "$ARTIFACT_DIR/$filters_to_apply"
-        cd "$ARTIFACT_DIR/$filters_to_apply"
-
-        step_idx=$((step_idx + 1))
-        VISU_LOG_FILE="$ARTIFACT_DIR/$filters_to_apply/${step_idx}_matbench_visualize.log"
-
-        export MATBENCH_FILTERS="$filters_to_apply"
-        if [[ -d "$MATBENCH_RESULTS_DIRNAME/lts" ]]; then
-            LTS_DIR="--lts_results_dirname $MATBENCH_RESULTS_DIRNAME/lts"
-        else
-            LTS_DIR=""
-        fi
-        if ! matbench visualize --generate="$generate_url" $LTS_DIR |& tee > "$VISU_LOG_FILE"; then
-            _warning "Visualization generation failed :("
-            retcode=1
-        fi
-        if grep "^ERROR" "$VISU_LOG_FILE"; then
-            _warning "An error happened during the report generation :/."
-            grep "^ERROR" "$VISU_LOG_FILE" >> "$ARTIFACT_DIR"/FAILURE
-            retcode=1
-        fi
-        unset MATBENCH_FILTERS
-
-        mkdir -p figures_{png,html}
-        mv fig_*.png "figures_png" 2>/dev/null || true
-        mv fig_*.html "figures_html" 2>/dev/null || true
-    done
-
-    cd "$ARTIFACT_DIR"
-    return $retcode
+        }' "$secret_file" >.env.generated.yaml
+	}
+
+	#
+	# Download the LTS results
+	# Analyze the current results against LTS results
+	#
+
+	if test_config matbench.lts.regression_analyses.enabled; then
+		generate_opensearch_config
+
+		step_idx=$((step_idx + 1))
+
+		if ! matbench download_lts --lts_results_dirname "$MATBENCH_RESULTS_DIRNAME/lts" |&
+			tee >"$ARTIFACT_DIR/${step_idx}_matbench_download_lts.log"; then
+			_warning "An error happened while downloading the LTS results :/."
+			retcode=1
+		fi
+
+		step_idx=$((step_idx + 1))
+		regression_analyses_retcode=0
+		matbench analyze-lts --lts_results_dirname "$MATBENCH_RESULTS_DIRNAME/lts" |&
+			tee >"$ARTIFACT_DIR/${step_idx}_matbench_analyze_lts.log" ||
```

Comment:
```suggestion
			tee > "$ARTIFACT_DIR/${step_idx}_matbench_analyze_lts.log" ||
```
I try to always keep a space before/after the `>` (except if redirecting into a file description, like `>&2`), let's avoid changing it here